datalad/distribution/get.py

```diff
@@ -0,0 +1,197 @@
+            doc="""specify the dataset to perform the add operation on.  If
+            no dataset is given, an attempt is made to identify the dataset
+            based on the current working directory and/or the `path` given""",
+            constraints=EnsureDataset() | EnsureNone()),
+        path=Parameter(
+            args=("path",),
+            metavar="PATH",
+            doc="""path/name of the requested dataset component. The component
+            must already be known to the dataset.""",
+            nargs="*",
+            constraints=EnsureStr() | EnsureNone()),
+        source=Parameter(
+            args=("-s", "--source",),
+            metavar="LABEL",
+            doc="""label of the data source to be used to fulfill the request.
+            This can be the name of a dataset :term:`sibling` or another known
+            source""",
+            constraints=EnsureStr() | EnsureNone()),
+        recursive=recursion_flag,
+        recursion_limit=recursion_limit,
+        git_opts=git_opts,
+        annex_opts=annex_opts,
+        annex_get_opts=annex_get_opts)
+
+    @staticmethod
+    @datasetmethod(name='get')
+    def __call__(
+            path=None,
+            source=None,
+            dataset=None,
+            recursive=False,
+            recursion_limit=None,
+            git_opts=None,
+            annex_opts=None,
+            annex_get_opts=None):
+
+        # resolve dataset:
+        ds = require_dataset(dataset, check_installed=True,
+                             purpose='getting content')
+
+        # check parameters:
+        if path is None:
+            raise InsufficientArgumentsError("insufficient information for "
+                                             "getting: requires at least a "
+                                             "path.")
+        # When called from cmdline `path` will be a list even if
+        # there is only one item.
+        # Make sure we deal with the same when called via python API:
+        if not isinstance(path, list):
+            path = [path]
+
+        # resolve path(s):
+        resolved_paths = [resolve_path(p, dataset) for p in path]
+
+        # resolve associated datasets:
+        resolved_datasets = dict()
+        for p in resolved_paths:
+            p_ds = ds.get_containing_subdataset(p,
+                                                recursion_limit=recursion_limit)
+            if p_ds is None:
+                raise FileNotInRepositoryError(
+                    msg="{0} not in dataset.".format(p))
+            if not recursive and p_ds != ds:
+                raise FileNotInRepositoryError(
+                    msg="{0} belongs to subdataset {1}. To get its content use "
+                        "option `recursive` or call get on the "
+                        "subdataset.".format(p, p_ds))
+            if p_ds.path not in resolved_datasets:
+                resolved_datasets[p_ds.path] = []
+            resolved_datasets[p_ds.path].append(p)
```

Comment:
Well, it's nicer, I agree.
But: How is `dict.get()` not a dict search? Or where do you think to save one?